plugins/ingest/src/test/java/org/elasticsearch/ingest/PipelineTests.java

```diff
@@ -17,20 +17,18 @@
     public void testProcessorSettingsRemainUntouched() throws Exception {
         Map<String, Object> subField = new HashMap<>();
         subField.put("_subfield", "value");
-        Map<String, Object> fieldSettings = new HashMap<>();
-        fieldSettings.put("_field", subField);
-        Map<String, Object> addSettings = new HashMap<>();
-        addSettings.put("fields", fieldSettings);
+        Map<String, Object> setSettings = new HashMap<>();
+        setSettings.put("_field", subField);
         Map<String, Object> removeSettings = new HashMap<>();
-        removeSettings.put("fields", Collections.singletonList("_field._subfield"));
-        Pipeline pipeline = createPipeline(processorConfig(SetProcessor.TYPE, addSettings), processorConfig(RemoveProcessor.TYPE, removeSettings));
+        removeSettings.put("field", "_field._subfield");
+        Pipeline pipeline = createPipeline(processorConfig(SetProcessor.TYPE, setSettings), processorConfig(RemoveProcessor.TYPE, removeSettings));
 
         IngestDocument ingestDocument = new IngestDocument("_index", "_type", "_id", new HashMap<>());
         pipeline.execute(ingestDocument);
 
         assertThat(ingestDocument.getSource().get("_field"), Matchers.notNullValue());
         assertThat(((Map) ingestDocument.getSource().get("_field")).get("_subfield"), Matchers.nullValue());
-        assertThat(((Map) fieldSettings.get("_field")).get("_subfield"), Matchers.equalTo("value"));
+        assertThat(((Map) setSettings.get("_field")).get("_subfield"), Matchers.equalTo("value"));
```

Comment:
it is expected that this fails, cause factories remove what they read no? before we had an additional level of wrapping, fields, so fields would get removed but the original map, containing the fields wouldn't be touched.